impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js

```diff
@@ -420,7 +420,9 @@ define(["dojo/_base/declare", "dijit/_WidgetBase", "dijit/_TemplatedMixin", "dij
   _unescapeAmpIfExist: function(str) {
     if (Array.isArray(str)) {
       str.forEach(function(val) {
-        val.replace(/&amp;/g, '&');
+        if (typeof str === 'string' || str instanceof String) {
+          val.replace(/&amp;/g, '&');
```

Comment:
Unless `val` is actually a `String`, not common, replacing characters on a string returns a new string and thus have no side effect on the given array.
So I think you should do `str[index] = val.replace( ... );`, where `index` is the second argument to the `forEach`.

The same applies to the replace line below, when not in the forEach case.
Did this ever work? Are we using `String` objects — which is usually not good practice — for this to  work?